packages/z/zlmediakit/xmake.lua

```diff
@@ -0,0 +1,107 @@
+    add_deps("cmake")
+    --add_links("zlmediakit","zltoolkit","flv","mpeg","srt","jsoncpp","mov","ssl","crypto")     -- The SSL version might be too low, and specific linking order may be needed.
+
+    on_install(function (package)
+        local configs = {
+            "-DENABLE_API=" .. (package:config("ENABLE_API") and "ON" or "OFF"),
+            "-DENABLE_API_STATIC_LIB=" .. (package:config("ENABLE_API_STATIC_LIB") and "ON" or "OFF"),
+            "-DENABLE_ASAN=" .. (package:config("ENABLE_ASAN") and "ON" or "OFF"),
+            "-DENABLE_CXX_API=" .. (package:config("ENABLE_CXX_API") and "ON" or "OFF"),
+            "-DENABLE_FAAC=" .. (package:config("ENABLE_FAAC") and "ON" or "OFF"),
+            "-DENABLE_FFMPEG=" .. (package:config("ENABLE_FFMPEG") and "ON" or "OFF"),
+            "-DENABLE_HLS=" .. (package:config("ENABLE_HLS") and "ON" or "OFF"),
+            "-DENABLE_JEMALLOC_STATIC=" .. (package:config("ENABLE_JEMALLOC_STATIC") and "ON" or "OFF"),
+            "-DENABLE_MEM_DEBUG=" .. (package:config("ENABLE_MEM_DEBUG") and "ON" or "OFF"),
+            "-DENABLE_MSVC_MT=" .. (package:config("ENABLE_MSVC_MT") and "ON" or "OFF"),
+            "-DENABLE_MYSQL=" .. (package:config("ENABLE_MYSQL") and "ON" or "OFF"),
+            "-DENABLE_RTPPROXY=" .. (package:config("ENABLE_RTPPROXY") and "ON" or "OFF"),
+            "-DENABLE_SERVER=" .. (package:config("ENABLE_SERVER") and "ON" or "OFF"),
+            "-DENABLE_SRT=" .. (package:config("ENABLE_SRT") and "ON" or "OFF"),
+            "-DENABLE_TESTS=" .. (package:config("ENABLE_TESTS") and "ON" or "OFF"),
+            "-DENABLE_SCTP=" .. (package:config("ENABLE_SCTP") and "ON" or "OFF"),
+            "-DENABLE_WEBRTC=" .. (package:config("ENABLE_WEBRTC") and "ON" or "OFF"),
+            "-DENABLE_X264=" .. (package:config("ENABLE_X264") and "ON" or "OFF"),
+            "-DENABLE_WEPOLL=" .. (package:config("ENABLE_WEPOLL") and "ON" or "OFF"),
+            "-DDISABLE_REPORT=" .. (package:config("DISABLE_REPORT") and "ON" or "OFF"),
+            "-DUSE_SOLUTION_FOLDERS=" .. (package:config("USE_SOLUTION_FOLDERS") and "ON" or "OFF")
+        }
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        import("package.tools.cmake").install(package, configs)
+        
+        print("========================================================")
+        local ZLMediaKit_dir = package:installdir("include/ZLMediaKit/**");
+        local ZLToolKit_dir = package:installdir("include/ZLToolKit/src/**");
+
+        os.cp("**.a", package:installdir("lib"))
+        os.cp("**.so", package:installdir("lib"))
+        os.cp("**.dll", package:installdir("lib"))
+        os.cp("**.pdb", package:installdir("lib"))
+        os.cp("**.lib", package:installdir("lib"))
+
+        for _, dir in ipairs(os.dirs(ZLMediaKit_dir)) do
+            os.cp(dir,package:installdir("include"))
```

Comment:
? why?

Reply:
Include directory hierarchy needs to be preserved

Reply:
os.cp support it. `os.cp ("src/**.h", "/tmp/", {rootdir="src"})`

https://xmake.io/#/manual/builtin_modules?id=oscp